packages/peregrine/lib/hooks/useAwaitQuery.js

```diff
@@ -0,0 +1,16 @@
+import { useCallback } from 'react';
+import { useApolloClient } from '@apollo/react-hooks';
+
+export const useAwaitQuery = query => {
+    const apolloClient = useApolloClient();
+
+    return useCallback(
+        variables => {
+            return apolloClient.query({
```

Comment:
If this hook is the only thing we take from the spike I still like it as we can await queries now :P Any idea on the timeline for the awaitable apollo hook?

Also do you know if this works for mutations?

Reply:
There is another function `mutation` on the ApolloClient, but `useMutation` already returns a Promise, so shouldn't need to support in our custom hook.

Reply:
Another option we discussed was just using `useLazyQuery` and passing the callback function to the action ie:

```js
const [runQuery, result] = useLazyQuery(GET_USER_QUERY);

useEffect(() => { 
  getUserDetails(runQuery);
}, [getUserDetails, runQuery]);
```

Reply:
Discussed IRL: I would feel uncomfortable using this hook for anything but passing query invokers to Redux. I think hooks should be training developers to think in the render loop cycle, and adding async data fetching seems counter to that pattern.